plugins/modules/slack.py

```diff
@@ -289,18 +298,25 @@ def recursive_escape_quotes(obj, keys):
 
 
 def build_payload_for_slack(text, channel, thread_id, username, icon_url, icon_emoji, link_names,
-                            parse, color, attachments, blocks, message_id):
+                            parse, color, attachments, blocks, message_id, prepend_hash):
     payload = {}
     if color == "normal" and text is not None:
         payload = dict(text=escape_quotes(text))
     elif text is not None:
         # With a custom color we have to set the message as attachment, and explicitly turn markdown parsing on for it.
         payload = dict(attachments=[dict(text=escape_quotes(text), color=color, mrkdwn_in=["text"])])
     if channel is not None:
-        if channel.startswith(('#', '@', 'C0', 'GF', 'G0', 'CP')):
-            payload['channel'] = channel
-        else:
+        if prepend_hash == 'auto':
+            if channel.startswith(('#', '@', 'C0', 'GF', 'G0', 'CP')):
+                payload['channel'] = channel
+            else:
+                payload['channel'] = '#' + channel
+        elif prepend_hash == 'always':
             payload['channel'] = '#' + channel
+        elif prepend_hash == 'never':
+            payload['channel'] = channel
+
+        payload['channel'] = channel
```

Comment:
```suggestion
```
This line now needs to be removed, otherwise the above `if`/`else` complex won't do anything :)

Reply:
:+1: